From 4181fc5183621bf5ebff31325d0b06f66c30d179 Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Wed, 20 May 2020 17:09:03 -0700
Subject: [PATCH 06/13] ls: port removed-directory test to NFS

* tests/ls/removed-directory.sh:
Port test to NFS, where one gets a stale file handle
when looking at a removed directory.
---
 tests/ls/removed-directory.sh | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/tests/ls/removed-directory.sh b/tests/ls/removed-directory.sh
index 63b209dee..319594907 100755
--- a/tests/ls/removed-directory.sh
+++ b/tests/ls/removed-directory.sh
@@ -21,13 +21,18 @@
 . "${srcdir=.}/tests/init.sh"; path_prepend_ ./src
 print_ver_ ls
 
-cwd=$(pwd)
 mkdir d || framework_failure_
-cd d || framework_failure_
-rmdir ../d || skip_ "can't remove working directory on this platform"
+(cd d || exit 10
+ rmdir ../d || exit 11
+ test -d . || exit 12
+ ls || exit 13) >out 2>err
+case $? in
+  11) skip_ "can't remove working directory on this platform";;
+  12) skip_ "can't examine removed working directory on this platform";;
+  13) fail=1;;
+  * ) framework_failure_;;
+esac
 
-ls >../out 2>../err || fail=1
-cd "$cwd" || framework_failure_
 compare /dev/null out || fail=1
 compare /dev/null err || fail=1
 
-- 
2.27.0

